base/services/w32time/w32time.c

```diff
@@ -279,29 +281,43 @@ W32TmServiceMain(DWORD argc, LPWSTR *argv)
     ServiceStatus.dwCurrentState = SERVICE_RUNNING;
     SetServiceStatus(hStatus, &ServiceStatus);
 
-    /* The worker loop of a service */
+    /* The service's worker loop */
     for (;;)
     {
-        result = SetTime();
+        /* Is it Auto-Sync? */
+        bAutoSync = FALSE;
+        if (RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                          L"SYSTEM\\CurrentControlSet\\Services\\W32Time\\Parameters",
+                          0,
+                          KEY_QUERY_VALUE,
+                          &hKey) == ERROR_SUCCESS)
+        {
+            cbData = sizeof(szData);
+            RegQueryValueExW(hKey, L"Type", NULL, NULL, (LPBYTE)szData, &cbData);
+            szData[ARRAYSIZE(szData) - 1] = UNICODE_NULL; /* Avoid buffer overrun */
+            bAutoSync = (wcscmp(szData, L"NTP") == 0);
 
-        if (result)
-            DPRINT("W32Time Service failed to set clock.\n");
-        else
-            DPRINT("W32Time Service successfully set clock.\n");
+            RegCloseKey(hKey);
+        }
 
-        if (result)
+        if (bAutoSync)
```

Comment:
This block should be inside the previous one.

Reply:
I don't think so.